packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/index.tsx

```diff
@@ -231,11 +231,12 @@ export default function HierarchyExplorer() {
         defaultCollapseIcon={<ExpandMoreIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
         defaultExpandIcon={<ChevronRightIcon sx={{ fontSize: '0.9rem', opacity: 0.5 }} />}
         expanded={Array.from(expandedDomNodeIdSet)}
-        selected={selectedDomNodeId as string}
+        selected={selectedDomNodeId}
         onNodeSelect={handleNodeSelect}
         onNodeFocus={handleNodeFocus}
         onNodeToggle={handleNodeToggle}
         onKeyDown={handleKeyDown}
+        multiSelect={false}
```

Comment:
cc @flaviendelangle There seems to be a change in the types of latest `@mui/x-tree-view` where the type of `onNodeSelect` expects `string | string[]` unless `multiSelect` is explicitly set to `false`.

Reply:
It should not be the case, I'll have a look

Reply:
Fixes in https://github.com/mui/mui-x/pull/10478